# Imports

In [1]:
from tqdm import tqdm_notebook as tqdm
import matplotlib
import time
import csv

In [2]:
%matplotlib inline

import matplotlib
import numpy as np
import matplotlib.pyplot as plt

# Sample Hourly Data

## Count

In [3]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [4]:
count = 0
for i in csv_reader:
    count += 1
    
print count

2141026


from one set of hourly data we have 2,141,026 entries. This is more than enough data and the rest of it while cool, is not necessary to get a good start

## Field Names

In [5]:
csv_reader.fieldnames

['WBAN',
 'Date',
 'Time',
 'StationType',
 'SkyCondition',
 'SkyConditionFlag',
 'Visibility',
 'VisibilityFlag',
 'WeatherType',
 'WeatherTypeFlag',
 'DryBulbFarenheit',
 'DryBulbFarenheitFlag',
 'DryBulbCelsius',
 'DryBulbCelsiusFlag',
 'WetBulbFarenheit',
 'WetBulbFarenheitFlag',
 'WetBulbCelsius',
 'WetBulbCelsiusFlag',
 'DewPointFarenheit',
 'DewPointFarenheitFlag',
 'DewPointCelsius',
 'DewPointCelsiusFlag',
 'RelativeHumidity',
 'RelativeHumidityFlag',
 'WindSpeed',
 'WindSpeedFlag',
 'WindDirection',
 'WindDirectionFlag',
 'ValueForWindCharacter',
 'ValueForWindCharacterFlag',
 'StationPressure',
 'StationPressureFlag',
 'PressureTendency',
 'PressureTendencyFlag',
 'PressureChange',
 'PressureChangeFlag',
 'SeaLevelPressure',
 'SeaLevelPressureFlag',
 'RecordType',
 'RecordTypeFlag',
 'HourlyPrecip',
 'HourlyPrecipFlag',
 'Altimeter',
 'AltimeterFlag']

## Sample Entry

In [20]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [19]:
csv_reader.next()

{'Altimeter': '30.24',
 'AltimeterFlag': ' ',
 'Date': '20070501',
 'DewPointCelsius': '-1.0',
 'DewPointCelsiusFlag': ' ',
 'DewPointFarenheit': '30',
 'DewPointFarenheitFlag': ' ',
 'DryBulbCelsius': '6.0',
 'DryBulbCelsiusFlag': ' ',
 'DryBulbFarenheit': '43',
 'DryBulbFarenheitFlag': ' ',
 'HourlyPrecip': ' ',
 'HourlyPrecipFlag': ' ',
 'PressureChange': ' ',
 'PressureChangeFlag': ' ',
 'PressureTendency': ' ',
 'PressureTendencyFlag': ' ',
 'RecordType': 'AA',
 'RecordTypeFlag': ' ',
 'RelativeHumidity': ' 60',
 'RelativeHumidityFlag': ' ',
 'SeaLevelPressure': 'M',
 'SeaLevelPressureFlag': ' ',
 'SkyCondition': 'CLR',
 'SkyConditionFlag': ' ',
 'StationPressure': '21.56',
 'StationPressureFlag': ' ',
 'StationType': '0',
 'Time': '0350',
 'ValueForWindCharacter': ' ',
 'ValueForWindCharacterFlag': ' ',
 'Visibility': '10.00',
 'VisibilityFlag': ' ',
 'WBAN': '03011',
 'WeatherType': ' ',
 'WeatherTypeFlag': ' ',
 'WetBulbCelsius': '2.7',
 'WetBulbCelsiusFlag': ' ',
 'WetBulbFare

# Classification Types

In [24]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [25]:
classifications = {}
field = 'WindSpeed'

# loop through data rows
for row in tqdm(csv_reader):
    if not row[field] in classifications:
        classifications[row[field]] = 0
        
    classifications[row[field]] += 1
    
classifications

{'  ': 200,
 ' 0': 264055,
 ' 1': 5441,
 ' 2': 10420,
 ' 3': 131265,
 ' 4': 8559,
 ' 5': 136558,
 ' 6': 136831,
 ' 7': 126804,
 ' 8': 114741,
 ' 9': 103853,
 '10': 87978,
 '11': 77871,
 '113': 1,
 '12': 911,
 '13': 61011,
 '14': 51611,
 '15': 40576,
 '16': 32636,
 '17': 26619,
 '18': 19968,
 '19': 170,
 '20': 15219,
 '21': 11834,
 '22': 8733,
 '23': 6731,
 '24': 4857,
 '25': 3737,
 '26': 2729,
 '27': 8,
 '28': 2038,
 '29': 1592,
 '30': 1030,
 '31': 749,
 '32': 554,
 '33': 377,
 '34': 266,
 '36': 189,
 '37': 145,
 '38': 110,
 '39': 68,
 '40': 83,
 '41': 47,
 '43': 48,
 '44': 40,
 '45': 34,
 '46': 27,
 '47': 15,
 '48': 24,
 '49': 14,
 '51': 18,
 '52': 20,
 '53': 9,
 '54': 9,
 '55': 13,
 '56': 6,
 '57': 10,
 '59': 6,
 '60': 6,
 '61': 7,
 '62': 4,
 '63': 4,
 '64': 6,
 '66': 3,
 '67': 3,
 '68': 2,
 '70': 3,
 '71': 1,
 '72': 1,
 '77': 1,
 'M': 641517}

# Format Data

In [127]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [128]:
bad_indexes = []
for row in tqdm(csv_reader):
    values = row.values()
    
    for i in range(len(values)):
        try:
            float(values[i])
        except:
            if i not in bad_indexes:
                bad_indexes.append(i)
            
len(bad_indexes)

40

In [129]:
csv_reader = csv.DictReader(open('./WeatherDirs/QCLCD200705/200705hourly.txt', 'r'))

In [130]:
data    = []
answers = []

for row in tqdm(csv_reader):
    if row[field].isdigit():
        answers.append(int(row[field]))

        data_vals = row.values()

        # delete classification field with answer
        for i in range(len(bad_indexes), 0, -1):
            del data_vals[i]

        data.append(data_vals)

# Machine Learning Test

In [131]:
from sklearn import svm

## Split Length

In [132]:
train_data = data[:(int(len(data) * .75))]
train_answ = answers[:(int(len(data) * .75))]

In [133]:
test_data = data[(int(len(data) * .75)):]
test_answ = answers[(int(len(data) * .75)):]

## Fit Data

In [134]:
clf = svm.SVC()
clf.fit(train_data, train_answ)  

ValueError: could not convert string to float: 

## Test Model